packages/core/src/deployment.ts

```diff
@@ -22,6 +22,10 @@ export interface Deployment {
   txHash?: string;
 }
 
+export interface DeploymentId {
+  deploymentId?: string;
```

Comment:
What's the reason not to add this directly in `Deployment`?

Reply:
There are some cases where we use `Required<Deployment>` to enforce that a deployment has a `txHash` defined.

For example, `deployProxy` previously used `Required<ProxyDeployment & DeployTransaction>` where `ProxyDeployment` extends `Deployment`.

We want the `deploymentId` to be optional, because it is only relevant when using Platform deploys.  So it cannot be part of `Deployment` if the above pattern continues to be used.

So now, `deployProxy` can use `Required<ProxyDeployment & DeployTransaction> & RemoteDeploymentId;`